### ISSUE TITLE
Back Button doesn't work

### ISSUE THREAD

**Issue by parkkralle** [home-assistant/frontend#17820]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

The back button does not work correctly (yout have to click it 5 times) after opening the more info dialog by clicking the value in an entities card.
If you click on the name or icon, it works. 

![image](https://github.com/home-assistant/frontend/assets/59173472/31a28f58-3a43-4fce-81de-473ff0db5953)
![image](https://github.com/home-assistant/frontend/assets/59173472/276aa428-2d10-4191-9232-b4d1bf9c75dd)

### Describe the behavior you expected

The Back Button should work correctly.

### Steps to reproduce the issue

1. Open a page with a back button 
2. Click on an Value in a entities card
3. Click on the back button 

### What version of Home Assistant Core has the issue?

2023.8.4

### In which browser are you experiencing the issue with?

Edge, Chrome, iOS App, Android App

### Which operating system are you using to run this browser?

Win 11, iOS, Android

**Comment by ildar170975** [comment]
Cannot confirm.
Win10+Chrome
HA2023.8.0

Steps:
1. Subview with Entities card is displayed.
2. Tap on a state of some entity.
3. more-info is displayed.
4. Instead of closing the more-info popup, tap on the "Back" button.
5. The more-info popup is closed, the "back" surf is not performed (as it is expected).
6. Tap on the "Back" button.
7. The "back" surf is performed.

![m](https://github.com/home-assistant/frontend/assets/71872483/835158c3-5d65-4e66-8672-a62fbb04b137)

**Comment by Xitee1** [comment]
Duplicate of #14571

**Comment by parkkralle** [comment]
Back button works correctly (More Info dialog opened by icon):

https://github.com/home-assistant/frontend/assets/59173472/293337b9-08b6-47cf-aeb5-cfd04a00de89


Back button does not work correctly (More Info dialog opened by value):

https://github.com/home-assistant/frontend/assets/59173472/ea673442-33a0-4e3b-90ac-e4ce492d7d34

**Comment by Xitee1** [comment]
Oh so that's why it sometimes works and sometimes not. It's exactly the same for me. When opening by pressing on the name, it works. When opening by pressing the value I need to press the back button multiple times.

**Comment by Xitee1** [comment]
Still an issue.

**Comment by KmanOz** [comment]
Yes an issue for me as well. What funny is I installed DuckDuck go browser on my Mac. Back button worked fine a few times and then just stopped working. Definitely an issue.

**Comment by KmanOz** [comment]
So I can add this. For anyone who is having this issue, please check this and see if the same happens on your system. First confirm the back button doesn't work. Then log out of HASS and log back in. As soon as I log back in the button works a few times and then reverts back to not working. Log out again and it will repeat. Strange indeed.

**Comment by pepisgo** [comment]
@KmanOz I tried that, it looks like it works only one time after logout/login.

**Comment by ildar170975** [comment]
Managed to reproduce it in different setups.
It is very simple:

1. Select a view with a `action: navigate` defined:
```
title: view_0
path: view_0
cards:
  - type: entities
    entities:
      - entity: sun.sun
        tap_action:
          action: navigate
          navigation_path: /lovelace-yaml/view_1
```
![image](https://github.com/home-assistant/frontend/assets/71872483/6f52097b-dce4-418a-a71b-49be99bf88c3)

2. Go to a subview:
```
title: view_1
path: view_1
subview: true
cards:
  - type: entity
    entity: zone.home
```
![image](https://github.com/home-assistant/frontend/assets/71872483/2f7abda0-21ee-4a51-9bbe-fcd0299379f2)


3. Tap on an entity to get a more-info popup:

![image](https://github.com/home-assistant/frontend/assets/71872483/01ce7174-260c-40ac-b7ec-a25fdb6efa6a)

**Note that so far you can close more info & press a back button -> you will leave the subview.**

4. Tap on "show more" - now you are on a History view:

![image](https://github.com/home-assistant/frontend/assets/71872483/3afc93bd-5e03-4704-b55f-0954141c1c5e)

5. Go back to more-info.
6. Close more-info -> you are on a subview.
7. Press the back button.

**Congratulations: you are not going back.**

In case exactly that sequence you will need TWO tapping the back button to go back.
But if you opening more-info & a history several times - then number of these tappings may be rather big.

**Comment by ildar170975** [comment]
And the glitch is also for a "Device" view:

1. Select any device with at least 1 entity. Assume this is a "mobile_app" device. Then a "Device" view is displayed:

![image](https://github.com/home-assistant/frontend/assets/71872483/27f97735-5cde-4e6c-8a51-83a9ddf95e9d)

2. Tap on any entity. A "more-info" popup is displayed:

![image](https://github.com/home-assistant/frontend/assets/71872483/dbbf96e6-6bba-44bb-9ba8-d90a6a8178f7)

3. Tap on "Show more" to show a History view:

![image](https://github.com/home-assistant/frontend/assets/71872483/91ffd696-8432-4f79-87f3-238ef6611a50)

4. Tap on "Back" to return to "more-info". Close "more-info" by pressing "X".
5. Repeat steps 2..4.
6. Tap on "Back" ("arrow left") - you need to press it several times to go back.

So, the glitch occurs if a user opened a History view.

**Comment by RonaldRegister** [comment]
can confirm this issue is stil present in 2024.3.1 on multiple browsers.

**Comment by UplandJacob** [comment]
This happens to me ALL the time. IOS app

**Comment by Flappiee** [comment]
Can confirm this issue still exists. See it many times when i open a device and look at the history of different sensors.
Have to press the "backup" up to 5 times before it is realy going back.

**Comment by tomas-dorda** [comment]
Me too

**Comment by UplandJacob** [comment]
https://github.com/home-assistant/frontend/assets/123428589/8bac1a12-b839-426d-81f0-33a70afee29a

**Comment by Xitee1** [comment]
Not exactly the same problem but related: For a while now the Android back button in the App also no longer works properly. When opening an entity in the dashboard and clicking show more, then pressing the back button brings you back to the entity  overview as expected but pressing it again to close the dialog, it closes the whole App instead going back to the dashboard.

**Comment by RonaldRegister** [comment]
Issue still present in 2024.7.3

**Comment by magdogg** [comment]
Same issue here in 2024.8. Please fix it Nabu Casa...

**Comment by UplandJacob** [comment]
Also is really a problem in a subview with a single webpage card. The back button controls the webpage card until it gets through all the history (like the back button in a normal browser).

Suggested fixes for this particular problem:
If the currently shown page is the same as the page the card is programmed to, skip the rest of the history.
OR: separate back button for the webpage card.

This bug seems to be caused by so many different things. It seems like the back button goes through _every single page_ that you were on, it needs to be simplified.

**Comment by tbrasser** [comment]
Same here, it seems we cycle through any history paths / url params instead of actually going to the previous page (in the hierarchy)

You can keep tapping the back button, at some point it _will_ work.

**Comment by UplandJacob** [comment]
>  at some point it _will_ work

I’ve had it get stuck before. When a page automatically redirects you (like for a sign in), you hit back, then it redirects you again. _Then you’re stuck._ 

The only way to get out is to clear the frontend cache. You can’t get to the normal menu in a subview, so In this case on mobile that means you have to spam the bottom right corner when you open the app to click the settings icon. Then go to **debugging** -> **reset frontend cache**.

**Comment by neilv** [comment]
can confirm this issue is stil present in 2024.9.1 on multiple browsers / Android App

**Comment by ddanielww** [comment]
New to home assistant, very powerful software, really like it, but a working back button shouldn’t be to much to ask for.

Logout and back in and it works for a little while     .

create automation , select back, save nothing.

This bug is making the application nearly impossible to use for me, please fix.

**Comment by philbert** [comment]
Ha, I never noticed it was clicking on the value that was problem. 
I just observed it sometimes happened, and other times it didn't but it usually got worse the more stuff you clicked on.
Since it works fine when you click on the name it must be a relatively straight forward fix to make it work the same when you click on the value.

**Comment by armantaherian** [comment]
@philbert It's somewhat straightforward, somewhat not. I figured out the problem in my local system: the open-modal trigger occurs multiple times when you click on a value. This happens at least once on the *row* and another time on the *value*. Each has its own event listener, and both open the modal. So, there's one modal but dozens of `history.push`.

**Comment by philbert** [comment]
Could you point out where in the code this is happening? I don't have much javascript experience, but I might have a shot at trying to fix it.

**Comment by KipK** [comment]
Quick question to those having this issue ? Do you all have Broswer Mod extension activated ? 
Removing browser mod solve the back button problem here.

**Comment by elupus** [comment]
Nope.

**Comment by ddanielww** [comment]
Do not have browser mod installed and back but never works, so frustrating

**Comment by tomas-dorda** [comment]
did not help

**Comment by ildar170975** [comment]
> Removing browser mod solve the back button problem here.

Not using browser-mod. 
It is super-easy to reproduce the issue - https://github.com/home-assistant/frontend/issues/17820#issuecomment-1975766026

**Comment by neilv** [comment]
Remove browser_mod, no effect. 😥

**Comment by ddanielww** [comment]
Every part of the system is affected, create an automation, back doesn’t work. It’s literally bugged all over the system, hope this gets fixed soon.

**Comment by magdogg** [comment]
It's a very real bug that is making using the HA UI very irritating.

**Comment by KipK** [comment]
You're right I still have the bug. I got it worse with BrowserMod, so gave me the illusion it was solved.. :)

**Comment by magdogg** [comment]
Yea the back button should perform the action: Go back one page. Nothing else.

**Comment by ddanielww** [comment]
https://github.com/user-attachments/assets/ffcf1b00-022b-4295-96b1-e193433199eb

So this is by design?

The back button does not function correctly all over hone assistant

**Comment by KipK** [comment]
Also the back button from the interface shouldn't be linked to browser history like it is.

If you get to home assistant ui from a direct link to settings as example, there's no way to get back to the home.

**Comment by ddanielww** [comment]
Is anyone reviewing this bug? It’s making my time with home assistant unusable

**Comment by backslashV** [comment]
Same problem since forever. Are HA folks even aware of this thread? I didn't see any acknowledgements here.

**Comment by Rihan9** [comment]
There are few duplicates about this issue. I guess they are working on one of them. I Hope so. The front-end is pretty frustrating to use right now due to this issue. From mobile (Android) is even worse because the navigation just broke completely and you are forced to close the app and reopen It.

**Comment by ddanielww** [comment]
I find out it mad how the basics of the application don’t work like the back button.

The notification system doesn’t work, shows 1 new notification but doesn’t show until you close and full reload the application.

sorry going of topic , a working back button / a way to navigate is essential please fix.

**Comment by mousemat86** [comment]
Nabu, please can we get a fix for this.

**Comment by zulu64** [comment]
@frenck, kindly give this bug some TLC.

It makes subview "back button" functionality useless if we use any modal windows within that subview, e.g. more info windows, bubble card et al.

My 2 cents:
The problem isn't with the "back button" - it retraces the history as expected. It looks like each modal window pushes itself onto the tab history, but doesn't pop it on exit, essentially corrupting the history.

If correct, 2 solutions come to mind for core modal windows e.g. more info:

1. Do not push onto the tab history on entry, or...
2. Pop from tab history on exit

Hope this helps.

**Comment by frenck** [comment]
I am not a frontend dev m8...

Let not ping people randomly to get attention; it doesn't work and can only lead to annoyance (I know you mean well though).

I would, however, welcome any contribution that fixes the issue 👍

../Frenck

**Comment by zulu64** [comment]
> I am not a frontend dev m8...
> 
> Lets not ping people randomly to get attention; it doesn't work and can only lead to annoyance (I know you mean well though).
> 
> I would, however, welcome any contribution that fixes the issue 👍
> 
> ../Frenck

My bad @frenck , from your previous work and release presentations I thought you _were_ a dev.

How **can** we highlight or vote for this bug then? Any guidance is appreciated because the severity is high even if the priority appears low within Nabu.

In the meantime, I'll look into contributing a fix myself :+1:

**Comment by frenck** [comment]
> My bad @frenck , from your previous work and release presentations I thought you _were_ a dev.

I am, I'm just not a frontender. There are many moving parts to this project.

> How **can** we highlight or vote for this bug then? Any guidance is appreciated because the severity is high even if the priority appears low within Nabu.

I think you are not fully understanding that Home Assistant is an open source project. It relies on contributions from the community to exists; including new features and bug fixes.

Nabu Casa hires a bunch of devs to be able to work full time on Home Assistant. Most of that time goes into building cross-functional features, reviewing contributions, handling releases, keeping forums online, maintaining other communications channels and what not.

tl;dr: Home Assistant is an open source project :)

../Frenck

**Comment by ddanielww** [comment]
Settings > Automations > Click an existing automation and then try and select back.

Does not work :(, I don't think it's a problem with the path/sub screens, the whole thing is buggy.

https://github.com/user-attachments/assets/64dada33-d4f0-4569-9f13-7b27aa36ace1

**Comment by zulu64** [comment]
For those hitting this problem, there is a documented solution if you know that the back button should always return to a particular page. This solves most of my issues as my problem dashboard is one view and multiple subviews.

Refer to the `backpath variable` as documented in [HA subviews documentation](https://www.home-assistant.io/dashboards/views/#subview) to hard-code the return page.
The back button is therefore no longer affected by subview modal windows corrupting the tab history.

Example:
```
- title: In the garden
  path: garden
  subview: true
  back_path: /lovelace/homepage
```

Hope this helps.

**Comment by magdogg** [comment]
Not sure how this would be any sort of solution. The back button should go to the previous page, always.

**Comment by magdogg** [comment]
I should also add that sometime it actually works, briefly. Like a minute ago everything was as it should be and the back button did what it is supposed to do. 10 minutes later: same problem.

**Comment by ddanielww** [comment]
I am a developer but only in the learning stages but I am going to see what I can do.

This issue is driving me mad so decided to look through the development documents etc.

I enabled development mode and this issue goes away.

In development mode caching is disabled.

Can everyone reset the cache via the web browser and see if navigation works, at least the first time around.

If this works at least initially I then need to figure out what can be done about the cache data.

**Comment by sandervs1979** [comment]
Indeed clearing the cache (CTRL+F5) resolves the issue (temporarily).
What I found while trying to trigger the unwanted behavior (open an automation, go to traces and then push the back button -> it should go back to the automation design), is that once I had opened one Bubble card and closed it normally, the issue re-appeared. Opening a sub-view or a more-info dialog and normally closing it would not trigger the unwanted back-button behavior. In all cases, clearing the cache would resolve it again.
Hope this helps!

**Comment by elupus** [comment]
Easiest repro for me.
Device info screen on a Device
Open info on an entity
On that entity open more info and select the related device
You are back at the original device
Repeat once or twice
Back button os funky

**Comment by ddanielww** [comment]
The above works for me.

The main problem for me seems to be the bubble card, as soon as I open a bubble card I can’t select back.

How many of you guys are using the bubble cards ?

Might be worth logging with the developer of the bubble card

**Comment by backslashV** [comment]
I'm using bubble cards. The issue is way more prominent and annoying on mobile, Android in my case. Sometimes, due to the funky behavior, the app ends up exiting.

**Comment by ddanielww** [comment]
I can’t seem to replicate the issue if I remove bubble cards. This issue needs logging with the developer of the bubble card

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by ddanielww** [comment]
I can’t seem to replicate the issue if I remove bubble cards. This issue needs logging with the developer of the bubble card

**Comment by mousemat86** [comment]
confirming I use bubble cards. Only recently though, come to think of it – that's potentially when I started experiencing this super annoying problem.

**Comment by magdogg** [comment]
Wow, this might be it. I already posted in the bubble card thread in the forums.

**Comment by neilv** [comment]
Yes, removed bubble-card and "back button" works again.
________________________________

**Comment by backslashV** [comment]
> I'm not using bubble cards and have this problem for months, in the webbrowser and the Android app.
> 
> It happens after viewing the history of an entity. Always after viewing history of something.

This ^^^
I have this issue even if I don't open any bubble cards.

**Comment by ddanielww** [comment]
I can view history of temperature and other presence sensors this causes no issues for me.

The only problem I have is when bubble card is enabled.

Do you have any other custom cards , themes etc though be breaking this for you ?

**Comment by Xitee1** [comment]
> Wow, this might be it. I already posted in the bubble card thread in the forums.

I'm also having this problem for a long time and do not have / never had the bubble card. I've also tested it on my testing instance (HA OS) that doesn't even has HACS nor any community integrations or cards installed (pretty much a fresh install). Clearing the cache doesn't help (tried it in incognito mode and with STRG + F5) either.

**Comment by ddanielww** [comment]
Can you send a video of your issue, I’ll then see if I can replicate it doing the same thing

**Comment by Xitee1** [comment]
> https://private-user-images.githubusercontent.com/59173472/265645793-ea673442-33a0-4e3b-90ac-e4ce492d7d34.mp4?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyODY1MDEsIm5iZiI6MTcyOTI4NjIwMSwicGF0aCI6Ii81OTE3MzQ3Mi8yNjU2NDU3OTMtZWE2NzM0NDItMzNhMC00ZTNiLTkwYWMtZTRjZTQ5MmQ3ZDM0Lm1wND9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE4VDIxMTY0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZlYjcwYzlhY2E2Mjg2NWViYWE1NGUwNzdiYWEyMmNkM2U3MTc1MjM5MDc5OGJmYmMyZjEyZTAyYmVhNGY3ZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.s_eofZFxO-q-xSDL2ufEfsQafD2yVPQE0JrbQ4sVndY

It's basically the same thing as in the video from @parkkralle.
I opened the overview of all devices in settings, opened a device overview, clicked on a value of an sensor, closed the history graph by clicking on a area outside of the window and then tried to navigate back.

**Comment by ddanielww** [comment]
https://github.com/user-attachments/assets/a7dbeab8-d4e5-4a0b-abb7-24b72ce70c68

I created a subpage to mock your flow and it works on mine without any issues.

I have found different cards cause issues themselves and custom integration.

Mine is the bubble card yours might be some other conflict / buggy card.

**Comment by Xitee1** [comment]
> it works on mine without any issues.

That's why you are clicking on the sensor name and not on its value. That's why it only "sometimes" happens, because it depends on where you click (at least for the sensors, the bug exists at other places too).

**Comment by ddanielww** [comment]
You are correct, there is an issue when you click the value and not the name.

The issue I was facing was due to the bubble card but this is also an issue with the home assistant core components.

Thanks Eitee


https://github.com/user-attachments/assets/a58e0592-0744-450f-a8ec-8d4c981df8eb

**Comment by magdogg** [comment]
Okay so the back button really is broken then...

**Comment by silamon** [comment]
Has anyone been able to reproduce this bug while being in safe mode? [Safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) will rule out problems with unsupported custom resources.

**Comment by ddanielww** [comment]
> You are correct, there is an issue when you click the value and not the name.
> 
> The issue I was facing was due to the bubble card but this is also an issue with the home assistant core components.
> 
> Thanks Xitee1
> 
>  value.mp4

This is a problem with the core, tested on a fresh install

**Comment by Clooos** [comment]
Hi everyone, I've released a new Bubble Card beta that fixes the issue for those experiencing this problem when Bubble Card is installed. I'm really sorry for that!

https://github.com/Clooos/Bubble-Card/releases/tag/v2.3.0-beta.3

**Comment by mousemat86** [comment]
I believe that might a fix ... Any one else?

**Comment by xupefei** [comment]
I found that the issue is related to the graph visualization.

Here's a reliable way to reproduce it:

1. Go to "Settings", "Devices & services", then go to any device that has a property of some sort of numbers, such as battery level. **The important thing is that, the property must be visualized as this kind of graph**:
<img width="400" alt="Screenshot 2024-10-23 at 18 15 15" src="https://github.com/user-attachments/assets/7d53a646-26b7-4fd3-8052-4f14aefa59a3">

2. Close the pop-up window.
3. Find the Back "<-" button and click on it.
4. Boom! You just have reproduced it.

**Comment by philbert** [comment]
Don't lose hope! 

I think I have a partial fix at least so that now only 2 items seem to be pushed to the browser history instead of 4. I believe I can properly fix the issue so that just one item is pushed with a bit of feedback from some proper frontend webdevs.

🤞

**Comment by THEthatdude** [comment]
Since I didn't see it mentioned- I also get this issue when working anywhere in the HA instance, this includes automations, devices, integrations, scripts, etc. Like I'll be working on an automation and get stuck on the main automation screen because the back button won't take me back- this happens mostly on the HA companion app not too often on Firefox on pc.

**Comment by armantaherian** [comment]
@philbert this is not a good way to handle the problem, it needs a proper refactorization.

**Comment by twobeass** [comment]
still an open topic since over a year now ? couldnt believe it is that hard to fix...

**Comment by 335iguy** [comment]
This is happening to me as well. 
I'm on Core 2024.10.1
Sup 2024.11.2
OS 13.1

**Comment by philbert** [comment]
Like many others here, I assumed this was an easy fix, but it definitely is not. I spent a good 20+ hours and gave it a shot but I accept my failure. One of the things I discovered is that if I removed the value element from hui-generic-entity-row.ts entirely so that it did not even appear in the browser, there is still an invisible clickable element present, that when clicked will pop-up the modal and still cause the back-button issue.  

This is not a straight-forward right-click inspect element thing. If you've tried, you'll notice that right clicking anywhere in the entity row does not actually pop-up the right-click menu. You have to click slightly above or below the row in order to inspect-element and then fish around in the DOM to try and find where the actual html is. 

The real issue appears to be hidden away in some dynamic html/shadow-DOM stuff that is well beyond my ability and needs a proper web developer to fix.

**Comment by arthanoss** [comment]
PR #22975 fixes back button after clicking on sensor value.

The underlying issue is however that the history isn't properly managed when a dialog is being replaced by another one. That was the case when clicking on a sensor value opened the same dialog 4 times, hence 4 presses of the back button were needed afterwards.

But it's still an issue, for example, when clicking _ADD ENTRY_ button on an integration page (the integration list dialog is being replaced by configuration dialog), or as seen [here](https://github.com/home-assistant/frontend/issues/14571#issuecomment-1689670324).

I'll take a closer look at this.

**Comment by wociscz** [comment]
> But it's still an issue, for example, when clicking _ADD ENTRY_ button on an integration page

Why closed then? Miss-click? @MindFreeze

**Comment by MindFreeze** [comment]
It was auto closed by the PR merge

**Comment by KipK** [comment]
this will fix the browser back button, but I think UI back button shouldn't behaves like browser one. It just should get back to parent section.

**Comment by dafunkydan** [comment]
👍  Same Problem here.
Might want to add: When using mobile Devices it is way more painful, as you dont have the same Possibilities to navigate.

One really easy Way to reproduce it:
- Open an Automation. Press back - you will get back.
- Open an Automation. Add another Action, edit a Field in that Action. Press back, Press back, Press back, Press back - you will get back. It can easily sum up to like 50 necessary Back-Commands, til you get back, depending on your Actions on that View.

**Comment by rgjertsen** [comment]
> 👍 Same Problem here. Might want to add: When using mobile Devices it is way more painful, as you dont have the same Possibilities to navigate.
> 
I second this. It has been bugging me for over a year.

**Comment by andrazek** [comment]
Yes, this is very frustrating. I usually kill the app, no other way to go back.

**Comment by elupus** [comment]
> Yes, this is very frustrating. I usually kill the app, no other way to go back.

You can usually stab the button a "billion" times to get it to go back. So it feels like something recursively fills the navigation list.

**Comment by MindFreeze** [comment]
I think I've fixed these issues with the last PR but it is tricky to find all the cases. So if this issue is still present on the next release, please reopen and add steps to reproduce it.

**Comment by Nazze** [comment]
It seems to be fixed now for me! (Was it with 12.3? I can't see the release in the PR).. Thanks a lot for your effort @MindFreeze !

**Comment by svh1985** [comment]
Having issue on iOS companion app that back button doesn't work.

Maybe a work around is to have a "deeper" pull to refresh. So if you pull down lower, the in app browser will navigate to the default Home Assistant page.

**Comment by MindFreeze** [comment]
> In my experience it got better but you still have to press the back button two times instead of only once to go back. (that is after opening a pop up bubble card)
> 
> EDIT: I'm using it with the Bubble Card

Custom cards are not relevant here. They should handle their own navigation or at least fire the proper events and follow our changes

> Having issue on iOS companion app that back button doesn't work.

Please provide specific steps to reproduce. We've been fixing specific cases of non compliant dialogs recently but there could be more.

**Comment by jsapede** [comment]
hi all,
looks like i'm also facing the problem 
whenever i make some changes on autom / scripts, cant get back using the button. 
is there a way to investigate this ?

**Comment by Flappiee** [comment]
I'm also facing this problem again. It was fixed however the bug has returned. Please fix it again.

**Comment by jsapede** [comment]
> > hi all, looks like i'm also facing the problem whenever i make some changes on autom / scripts, cant get back using the button. is there a way to investigate this ?
> 
> > Please provide specific steps to reproduce

its hard to find a recurrent step as its used for navigation and when it.comes i dont exactly remember the steps

will have a try

**Comment by jsapede** [comment]
> > hi all, looks like i'm also facing the problem whenever i make some changes on autom / scripts, cant get back using the button. is there a way to investigate this ?
> 
> > Please provide specific steps to reproduce

ok, here's what i see

go to strings, open intégration, or automation, back button works

witout leaving thè setting panel you're on, Switch companion app in the background, like to go on chrome or github, then switch back to companion app, and back button becomes unresponsive. all other fonctions work, you can édit your automation for exemple

tried disabling battery optimisation on Android bit has no effect. shouldnt be linked since other fonctions works, only back button is affected

**Comment by jsapede** [comment]
regularly having this error on logs, cant find why, maybe its linked

![Screenshot_20250614-082941.png](https://github.com/user-attachments/assets/a4babd95-2779-4cbe-b397-0843b5c4ca00)

**Comment by TimoPtr** [comment]
🤔 recently we've fixed `Show first view when opening the app` https://github.com/home-assistant/android/pull/5324 This feature actually navigate to the main dashboard and clear the history. @jsapede Did you by any chance use this feature if so could you try to disable it and try again.

If this is the case it means that the navigate function doesn't work in some cases and let the user on the previous screen with no way to navigate back since the history has been removed. 

Also when I'm trying to reproduce it, when I'm on the settings I only have the system back button is it the one you are referring @jsapede (gesture or navigation bar button)?

**Comment by jsapede** [comment]
looks like disabling first view solves the problem

**Comment by jsapede** [comment]
![Screenshot_20250616-103529.png](https://github.com/user-attachments/assets/9ad6a9a4-bde9-4e9d-9b1c-894c4e9edd40)

this back button, as well as the gestures
